Monika After Story/game/definitions.rpy

```diff
@@ -5546,7 +5555,64 @@ init 1 python in mas_randchat:
 
         return randchat_disp
 
+        
+    def startWaiting():
+        """
+        Sets up the waiting time for the next random chat, depending on the current random chatter selection.
+        """
+        global rand_chat_waittime_left
+        
+        rand_chat_waittime_left = renpy.random.randint(rand_low, rand_high)
+        
+        
+    def wait():
+        """
+        Pauses renpy for a max of 5 seconds.
+        This helps adapting fast to a new random chatter selection.
+        """
+        global rand_chat_waittime_left
+        
+        if rand_chat_waittime_left > 5:
+            rand_chat_waittime_left -= 5
+            renpy.pause(5, hard=True)
+            
+        elif rand_chat_waittime_left > 0:
+            waitFor = rand_chat_waittime_left
+            rand_chat_waittime_left = 0
+            renpy.pause(waitFor, hard=True)
+            
+        else:
+            rand_chat_waittime_left = 0
+            renpy.pause(5, hard=True)
+                
+        
+    def waitedLongEnough():
+        """
+        Checks whether the waiting time is up yet.
 
+        RETURNS:
+            boolean to determine whether the wait is over
+        """
+        global rand_chat_waittime_left
+        
+        return rand_chat_waittime_left == 0 and rand_low != 0
+       
+       
+    def startWaitProcess():
+        """
+        Use this method for convenience.
+        It can be used to actively wait the random chatter time.
+        """
+        startWaiting()
+        
+        while not waitedLongEnough():
```

Comment:
so if random chatter is set to NEVER, `waitedLongEnough` always returns False and this loop constantly waits(). This prevents idle operations in `ch30_loop` (script-ch30) from running (Event checking / delayedactions / seasonalchecks / filereactions / persistent saving / affection saving)

Adding a check in this function for the NEVER case where you wait, then jump to the `post_pick_random_topic` label could solve this issue.

Reply:
I suppose you mean the preloop needs to be checked constantly?

Reply:
I wasn't sure about this, I'll remove the while loop to keep it checked then.

Reply:
the `ch30_loop` label, down where it does a thing to `calendar_last_checked`. All of that is designed to run once per minute, but requires a jump to `ch3_loop` every so often.

Reply:
yeah I think removing the while loop might work if you instead jump to the `post_pick_random_topic` label if you haven't waited long enough.